collect_data.sh

```diff
@@ -36,6 +36,23 @@ function get_running_time_from_user {
 
 }
 
+function get_ip_xor_mask_from_user {
+        user_ip_xor_mask=""
+        validated_ip_mask=false
+        while ! $validated_ip_mask; do
+                if [ -z "$user_ip_xor_mask" ]; then
+                        read -p "Please enter an ip mask for using to xor the original ips: " user_ip_xor_mask
+                        continue
+                fi
+                is_valid_ip=$(echo $user_ip_xor_mask |  grep -oE $ip_regex_validation)
+                if [ -z "$is_valid_ip" ]; then
+                        read -p "Please enter an ip mask for using to xor the original ips: " user_ip_xor_mask
```

Comment:
```suggestion
                        read -p "Please enter a string that will be used to mask IPs: " user_ip_xor_mask
```

Reply:
It's not a string, it's ip that with it i am doing the mask, for example 1.1.1.1 and one of the servers ip is 1.2.3.4 the n it will do xor between this 2 ip's. It will do it by the bits place - 1^1 . 1^2. 1^3. 1^4